src/components/pages/deals/Deals.js

```diff
@@ -1,33 +1,65 @@
 import React, { useState, useEffect } from 'react'
 
-import itemsServices from '../../../services/items'
+import fetchItemList from '../../../services/items'
 import ProductsOnDeals from '../../products/Products'
 import SearchBar from '../../searchbar/Searchbar'
 
 import './deals.css'
+import Pagination from '../../pagination/Pagination'
+import constants from '../../../utils/constants'
 
 const Deals = () => {
     const [deals, setDeals] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const [totalItems, setTotalItems] = useState(null)
+    const [page, setPage] = useState(1)
+    const [search, setSearch] = useState('')
 
     useEffect(() => {
         const fetch_deals = async () => {
-            const query = 'isOnSale=true'
-            const response_data = await itemsServices.getDeals(query)
-            setDeals(response_data.items)
+            // Calculate the "from" of the query params 
+            const request_from = (page - 1) * constants.PAGE_SIZE
+            // Query made to look for deals and from a certain number
+            const params = {
+                isOnSale: true,
+                from: request_from,
+                q:search
+            }
+            const response_deals = await fetchItemList(params)
+            // Set the received deals
+            setDeals(response_deals.items)
+            // set number of pages
+            setTotalItems(response_deals.total)
+            
             setIsLoading(false)
         }
+
         fetch_deals()
-    }, [])
+    }, [page, search])
```

Comment:
Beautiful! I love it! Awesome job with this useEffect :)

Reply:
Thanks for the encouragement. It pumps me up to want to do more😁